test/box/gh-5610-dirty-restart.lua

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env tarantool
+
+box.cfg{
+    listen                 = os.getenv("LISTEN"),
+    memtx_memory           = 107374182,
+    pid_file               = "tarantool.pid",
```

Comment:
Anyway didn't get the necessity of memtx_memory and pid_file settings. Otherwise LGTM